### PR TITLE
Clean up our post build step

### PR DESCRIPTION
**Pull request by jaredpar**
Infrastructure Change

This change has two effects to our build process:

1. Ports our GitLink fix from master
1. Creates an Index directory as a part of our post build work

Our index step today is very inefficient because it indexes virtually
every binary that Roslyn produces.  In actuality it should just be
indexing the ones that we ship.  The difference is literally gigs of
data.

The index diretory solves this by creating a simple layout which
contains only the data that actually needs to be indexed.  The index
step can simply target every file in this directory.

### CODE REVIEW

**Comment by jaredpar** [comment]
CC @jasonmalinowski, @tmat, @Pilchie 

Note: this is against the preview branch, not master.

File: src/Tools/MicroBuild/post-build.ps1 [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    Write-Host "Running GitLink"
+    foreach ($filePath in $binaryList) { 
+        $pdbPath = [IO.Path]::ChangeExtension($filePath, ".pdb")
+        Write-Host "`t$pdbPath"
+        Exec-Block { & $gitlink -u "https://github.com/dotnet/roslyn" $pdbPath --baseDir $repoDir } | Write-Host
+    }
+}
+
+# Create single directory to hold all of the items that we need to be indexed by
```
Comment (jasonmalinowski) [inline src/Tools/MicroBuild/post-build.ps1]: Is the intent of this to limit the indexing jobs we're shipping to the corporate symbol server? What impact does this also make for symbol archiving?

I guess I'm not exactly sure what "official build system" and "the indexing script" means here. My first read thought it meant *this* script which was really confusing.

File: src/Tools/MicroBuild/post-build.ps1 [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+#
+#   1. Run GitLink on the PDBs so we have proper source linking.
+#   2. Setup the <Configuration>\Index directory for simple / efficient indexing in official builds
+
+
+param([string]$config = "Release")
+Set-StrictMode -version 2.0
+$ErrorActionPreference="Stop"
+
+# Get the path to all of the shipping EXE / DLL files.
```
Comment (jasonmalinowski) [inline src/Tools/MicroBuild/post-build.ps1]: This should also include unit tests and other stuff too; we do want those to work.

Comment (jaredpar) [inline src/Tools/MicroBuild/post-build.ps1]: What is the point of indexing non-shipping binaries?

File: src/Tools/MicroBuild/post-build.ps1 (hunk 1 above)
Comment (jaredpar) [inline src/Tools/MicroBuild/post-build.ps1]: This concerns the MicroBuild Index Symbols step in our official build.  Today it is indexing virtually every binary that appears in our `Binaries\Release` folder.  Irrespective of whether or not we actually build or shipped it.  

Given the parameters to this build task it's not realistic to fix it with our present directory layout.  Much easier to create a simpler directory layout and then change that official build step to point at this new directory.

**Comment by jaredpar** [comment]
Closing after discussion.

**Comment by tmat** [comment]
@jaredpar What's the conclusion of the discussion?

**Pull request closed without merging**